packages/internal/src/utils.js

```diff
@@ -139,15 +144,20 @@ export const applyLabelingError = (func, args, label = undefined) => {
 };
 harden(applyLabelingError);
 
-const compareStringified = (left, right) => {
-  left = String(left);
-  right = String(right);
+/**
+ * @param {unknown} a
+ * @param {unknown} b
+ * @returns {-1 | 0 | 1}
+ */
+const compareStringified = (a, b) => {
+  const left = String(a);
+  const right = String(b);
   // eslint-disable-next-line no-nested-ternary
   return left < right ? -1 : left > right ? 1 : 0;
 };
 
 /**
- * @param {object} obj
+ * @param {Record<string | symbol, unknown>} obj
  * @returns {(string|symbol)[]}
```

Comment:
Would be neat to have an `infer` based result here, but can be left as future improvement.

Reply:
it's correctly inferred now. I don't see the value in removing the `returns` but open to it.

Reply:
I meant with some work we could type this utility function so that the it would actually have as return type the subset of the `obj` keys for which the values are functions. Aka a type implementation of the behavior.

Sure `(string|symbol)[]` is correct, but it's fairly broad.

Reply:
Ah got it. I think we'd run into this: https://stackoverflow.com/a/55012175/46040 (found from `https://github.com/microsoft/TypeScript/issues/41036`)